crypto/dh/dh_ameth.c

```diff
@@ -185,6 +187,7 @@ static int dh_priv_decode(EVP_PKEY *pkey, const PKCS8_PRIV_KEY_INFO *p8)
     if (!DH_generate_key(dh))
         goto dherr;
 
+    dh->dirty_cnt++;
```

Comment:
Do these need to be atomic?

Do we make thread safety claims for PKEY objects?

Reply:
There's only one lock in the DH code, and it's there to guard the reference counter.  If you grep for "lock" in crypto/evp/, you will see the same thing there.